enclave-runtime/src/lib.rs

```diff
@@ -139,25 +144,55 @@ pub unsafe extern "C" fn init() -> sgx_status_t {
 		return e.into()
 	}
 
-	// create the aes key that is used for state encryption such that a key is always present in tests.
-	// It will be overwritten anyway if mutual remote attastation is performed with the primary worker
+	// Create the aes key that is used for state encryption such that a key is always present in tests.
+	// It will be overwritten anyway if mutual remote attastation is performed with the primary worker.
 	if let Err(e) = aes::create_sealed_if_absent().map_err(Error::Crypto) {
 		return e.into()
 	}
 
 	let state_handler = GlobalFileStateHandler;
 
-	// for debug purposes, list shards. no problem to panic if fails
+	// For debug purposes, list shards. no problem to panic if fails.
 	let shards = state_handler.list_shards().unwrap();
 	debug!("found the following {} shards on disk:", shards.len());
 	for s in shards {
 		debug!("{}", s.encode().to_base58())
 	}
-	//shards.into_iter().map(|s| debug!("{}", s.encode().to_base58()));
+
+	let mu_ra_url =
+		match String::decode(&mut slice::from_raw_parts(mu_ra_addr, mu_ra_addr_size as usize))
+			.map_err(Error::Codec)
+		{
+			Ok(addr) => addr,
+			Err(e) => return e.into(),
+		};
+
+	let unstrusted_worker_url = match String::decode(&mut slice::from_raw_parts(
+		untrusted_worker_addr,
+		untrusted_worker_addr_size as usize,
+	))
+	.map_err(Error::Codec)
+	{
+		Ok(addr) => addr,
+		Err(e) => return e.into(),
+	};
+
+	if let Err(e) = set_primitives(&mu_ra_url, &unstrusted_worker_url) {
+		return e.into()
+	}
 
 	sgx_status_t::SGX_SUCCESS
 }
 
+fn set_primitives(mu_ra_url: &str, unstrusted_worker_url: &str) -> Result<()> {
+	let primitives = Primitives::new(mu_ra_url, unstrusted_worker_url);
+	let mut rw_lock = GLOBAL_PRIMITIVES_CACHE.load_for_mutation()?;
+
+	*rw_lock = primitives;
+
+	Ok(())
+}
```

Comment:
Is there a reason that this is in the enclave-runtime?

Reply:
Good question. Moved it to primitives-cache. Though, it might as well be removed completely.